plans/phase-1/final-cr-definition.md

```diff
@@ -0,0 +1,845 @@
+# Final Multigres Operator CR Definition
+
+```ascii
+[MultigresCluster] 🚀 (The root CR - user-editable)
+      │
+      ├── 🌍 [GlobalTopoServer] (Child CR if managed)
+      │    │
+      │    └── 🏛️ etcd Resources (if managed)
+      │
+      ├── 🤖 MultiAdmin Resources - Deployment, Services, Etc
+      │    
+      │
+      ├── 💠 [Cell] (Child CR)
+      │    │
+      │    ├── 🚪 MultiGate Resources (Deployment, Service, etc.)
+      │    │    
+      │    │
+      │    ├── 🧠 MultiOrch Resources (Deployment, etc.)
+      │    │    
+      │    │
+      │    └── 📡 [LocalTopoServer] (Child CR if managed and not using global)
+      │         │
+      │         └── 🏛️ etcd Resources (if managed)
+      │
+      └── 🗃️ [TableGroup] (Child CR)
+           │
+           └── 📦 [Shard] (Child CR)
+                │
+                └── 🏊 MultiPooler and postgres resources (pods or statefulset)
+                    
+
+
+📋 [DeploymentTemplate] (Separate CR - user-editable, NOT a child)
+   ├── Contains spec sections for:
+   │   ├── multiadmin
+   │   ├── multigateway
+   │   ├── multiorch
+   │   ├── shardPool
+   │   ├── images
+   │   └── managedTopoServer
+   │
+   └── Watched by MultigresCluster controller ONLY when referenced
+       └── Resolved into child CRs (children are unaware of templates)
+
+```
+
+## Multigres Cluster
+
+This and the DeploymentTemplate are the only two editable entries for the end-user. All other child CRs are read-only.
+
+```yaml
+apiVersion: multigres.com/v1alpha1
+kind: MultigresCluster
+metadata:
+  name: example-multigres-cluster
+  namespace: multigres
+spec:
+  # Images are defined globally to avoid the danger of running multiple incongruent versions at once.
+  # The operator will eventually take care of rolling updates in a safe way.
+  # Optional
+  images:
+    deploymentTemplate: "standard-ha"
+    # --- ALTERNATIVE: Inline Definition ---
+    # If 'deploymentTemplate' is omitted, the controller uses this spec directly.
+    # multigateway: "multigres/multigres:latest"
+    # multiorch: "multigres/multigres:latest"
+    # multipooler: "multigres/multigres:latest"
+    # multiadmin: "multigres/multigres:latest"
+    # postgres: "postgres:15.3"
+
+  # ----------------------------------------------------------------
+  # Base Cluster Configuration
+  # ----------------------------------------------------------------
+  # Optional
+  globalTopoServer:
+    rootPath: "/multigres/global"
+    deploymentTemplate: "standard-ha"
+    # --- ALTERNATIVE: Inline Definition ---
+    # If 'deploymentTemplate' is omitted, the controller uses this spec directly.
```

Comment:
<s>It is  common to use the template input as the "base" input, and override with some inline definitions -- but I can see there is CEL for mutually exclusive setup. It should be clarified in this document that it's not allowed to use override.</s>

I thought about adding the above comment, but actually there seems to be "overrides" field to achieve exactly that. It should be mentioned here as well.